proxy/envoy/config_test.go

```diff
@@ -216,31 +216,100 @@ func TestTCPRouteConfigByRoute(t *testing.T) {
 	}
 }
 
+type fileConfig struct {
+	meta model.ConfigMeta
+	file string
+}
+
 const (
 	envoyConfig = "testdata/envoy.json"
+)
+
+var (
+	cbPolicy = fileConfig{
+		meta: model.ConfigMeta{Type: model.DestinationPolicy.Type, Name: "circuit-breaker"},
+		file: "testdata/cb-policy.yaml.golden",
+	}
+
+	timeoutRouteRule = fileConfig{
+		meta: model.ConfigMeta{Type: model.RouteRule.Type, Name: "timeout"},
+		file: "testdata/timeout-route-rule.yaml.golden",
+	}
+
+	weightedRouteRule = fileConfig{
+		meta: model.ConfigMeta{Type: model.RouteRule.Type, Name: "weighted"},
+		file: "testdata/weighted-route.yaml.golden",
+	}
+
+	faultRouteRule = fileConfig{
+		meta: model.ConfigMeta{Type: model.RouteRule.Type, Name: "fault"},
+		file: "testdata/fault-route.yaml.golden",
+	}
+
+	redirectRouteRule = fileConfig{
+		meta: model.ConfigMeta{Type: model.RouteRule.Type, Name: "redirect"},
+		file: "testdata/redirect-route.yaml.golden",
+	}
+
+	rewriteRouteRule = fileConfig{
+		meta: model.ConfigMeta{Type: model.RouteRule.Type, Name: "rewrite"},
+		file: "testdata/rewrite-route.yaml.golden",
+	}
 
-	cbPolicy              = "testdata/cb-policy.yaml.golden"
-	timeoutRouteRule      = "testdata/timeout-route-rule.yaml.golden"
-	weightedRouteRule     = "testdata/weighted-route.yaml.golden"
-	faultRouteRule        = "testdata/fault-route.yaml.golden"
-	redirectRouteRule     = "testdata/redirect-route.yaml.golden"
-	rewriteRouteRule      = "testdata/rewrite-route.yaml.golden"
-	websocketRouteRule    = "testdata/websocket-route.yaml.golden"
-	egressRule            = "testdata/egress-rule.yaml.golden"
-	egressRuleCBPolicy    = "testdata/egress-rule-cb-policy.yaml.golden"
-	egressRuleTimeoutRule = "testdata/egress-rule-timeout-route-rule.yaml.golden"
+	websocketRouteRule = fileConfig{
+		meta: model.ConfigMeta{Type: model.RouteRule.Type, Name: "websocket"},
+		file: "testdata/websocket-route.yaml.golden",
+	}
```

Comment:
nit: can we move the config metadata to the golden, so that it serves as a reference to others? is there some programmatic fu that we are doing or planning to do in future that would necessitate creating metadata programmatically?

Reply:
that's a workaround to avoid depending on CRD parsing code in proxy. the parser is borrowed from kubernetes, and I wanted to avoid indirect dependencies on kubernetes.